src/telegram/_messageautodeletetimerchanged.py

```diff
@@ -35,26 +41,52 @@ class MessageAutoDeleteTimerChanged(TelegramObject):
     .. versionadded:: 13.4
 
     Args:
-        message_auto_delete_time (:obj:`int`): New auto-delete time for messages in the
-            chat.
+        message_auto_delete_time (:obj:`int` | :class:`datetime.timedelta`): New auto-delete time
+            for messages in the chat.
+
+            .. versionchanged:: NEXT.VERSION
+                |time-period-input|
 
     Attributes:
-        message_auto_delete_time (:obj:`int`): New auto-delete time for messages in the
-            chat.
+        message_auto_delete_time (:obj:`int` | :class:`datetime.timedelta`): New auto-delete time
+            for messages in the chat.
+
+            .. deprecated:: NEXT.VERSION
+                |time-period-int-deprecated|
 
     """
 
-    __slots__ = ("message_auto_delete_time",)
+    __slots__ = ("_message_auto_delete_time",)
 
     def __init__(
         self,
-        message_auto_delete_time: int,
+        message_auto_delete_time: TimePeriod,
         *,
         api_kwargs: Optional[JSONDict] = None,
     ):
         super().__init__(api_kwargs=api_kwargs)
-        self.message_auto_delete_time: int = message_auto_delete_time
+        self._message_auto_delete_time: dtm.timedelta = parse_period_arg(
+            message_auto_delete_time
+        )  # type: ignore[assignment]
 
         self._id_attrs = (self.message_auto_delete_time,)
 
         self._freeze()
+
+    @property
+    def message_auto_delete_time(self) -> Union[int, dtm.timedelta]:
```

Comment:
```suggestion
    def message_auto_delete_time(self) -> Optional[TimePeriod]:
```

same for all other places. This should also resolve mypy complaining.